base/services/w32time/ntpclient.c

```diff
@@ -0,0 +1,175 @@
+/*
+ * PROJECT:     ReactOS Timedate Control Panel
+ * LICENSE:     GPL-2.0+ (https://spdx.org/licenses/GPL-2.0+)
+ * PURPOSE:     Queries the NTP server
+ * COPYRIGHT:   Copyright 2006 Ged Murphy <gedmurphy@gmail.com>
+ */
+
+#include "timedate.h"
+#include <winsock2.h>
+
+#define TIMEOUT 4000 /* 4 second timeout */
+
+typedef struct _INFO
+{
+    SOCKET Sock;
+    SOCKADDR_IN myAddr;
+    SOCKADDR_IN ntpAddr;
+    NTPPACKET SendPacket;
+    NTPPACKET RecvPacket;
+} INFO, *PINFO;
+
+
+static BOOL
+InitConnection(PINFO pInfo, LPSTR lpAddress)
+{
+    WSADATA wsaData;
+    HOSTENT *he;
+    INT Ret;
+
+    Ret = WSAStartup(MAKEWORD(2, 2), &wsaData);
+    if (Ret != 0)
+        return FALSE;
+
+    pInfo->Sock = socket(AF_INET,
+                         SOCK_DGRAM,
+                         0);
+    if (pInfo->Sock == INVALID_SOCKET)
+        return FALSE;
+
+    /* Setup server info */
+    he = gethostbyname(lpAddress);
+    if (he == NULL)
+        return FALSE;
+
+        /* Setup server socket info */
+        ZeroMemory(&pInfo->ntpAddr, sizeof(SOCKADDR_IN));
```

Comment:
Please use `sizeof(pInfo->ntpAddr)` here

Reply:
Copied from "dll/cpl/timedate/timedate.h" in its entirety and then I commented out a reference to "resource.h" because it was not needed here, but I will try and make the changes requested.